src/data/phase/dto.type.ts

```diff
@@ -20,11 +20,21 @@ export type TaskDTO = Readonly<{
   points: number
 }>
 
+export type ChallengeContent = Readonly<{
+  challenge: string
+}>
+
+export type ChallengeDTO = Readonly<{
```

Comment:
You miss some elements to be more extensible.
For now we only handle part one of the challenges, but how will you do for the part two and so on?
You need to design the `challengeDTO` in the same way of the tasks to achieve that.